Source/WebCore/animation/KeyframeEffectStack.cpp

```diff
@@ -68,31 +68,34 @@ void KeyframeEffectStack::removeEffect(KeyframeEffect& effect)
         startAcceleratedAnimationsIfPossible();
 }
 
-bool KeyframeEffectStack::requiresPseudoElement() const
+bool KeyframeEffectStack::hasMatchingEffect(const Function<bool(const KeyframeEffect&)>& function) const
```

Comment:
Why do you need this at all? Can't you just have the call sites use Vector's `findIf()` ?

Reply:
I didn't know about `findIf()` even though I went looking for it. More often than not when I look for something in `String` or `Vector` I end up not finding it…

That said, I wonder if using `findIf()` in this case will be so nice because the values are `WeakPtr` and using the iterator skips any potential null value, as I understand it. I'm not sure that `findIf()` will do that.